install_config/cluster_metrics.adoc

```diff
@@ -753,3 +753,291 @@ by performing the following steps:
+* Prometheus and alertmanager configmaps.
+* Prometheus and alerts services and direct routes.
+* Prometheus, alertmanager and alert-buffer PVCs.
+
+Depends on `openshift_hosted_prometheus_deploy` flag.
+
+Set the following role variables to install and configure Prometheus.
+
+.Prometheus Variables
+[options="header"]
+|===
+
+|Variable |Description
+
+|`openshift_prometheus_state`
+|Set variable to install and update. Remove variable to uninstall.
+
+|`openshift_prometheus_namespace`
+|Project namespace where the components will be deployed. Default set to
+`openshift-prometheus`.
+
+|`openshift_prometheus_replicas`
+|The number of replicas for prometheus deployment. The default is one. Multiple
+replicas are not supported.
+
+|`openshift_prometheus_node_selector`
+|Selector for the nodes on which prometheus will be deployed.
+
+|`openshift_prometheus_image_proxy`
+|Set to specify image for the component. For example, "openshift/oauth-proxy:v1.0.0".
+
+|`openshift_prometheus_image_prometheus`
+|Set to specify image for the component. For example, "openshift/prometheus:v2.0.0-dev".
+
+|`openshift_prometheus_image_alertmanager`
+|Set to specify image for the component. For example, "openshift/prometheus-alertmanager:dev".
+
+|`openshift_prometheus_image_alertbuffer`
+|Set to specify image for the component. For example, "ilackarms/message-buffer".
+
+|`openshift_prometheus_additional_rules_file`
+|Additional Prometheus rules file. Set to `null` by default.
+
+|===
+
+[openshift-prometheus-deploy]
+=== Deploying Prometheus Using Ansible Installer
+
+The Ansible Installer is the default method of deploying Prometheus.
+
+Add label to your node:
+----
+region=infra and deploy prometheus via ansible with default values
+# default value inventory file
+openshift_prometheus_state=present
+openshift_prometheus_namespace=prometheus
+
+openshift_prometheus_replicas=1
+openshift_prometheus_node_selector={"region":"infra"}
+
+# images
+openshift_prometheus_image_proxy="openshift/oauth-proxy:v1.0.0"
+openshift_prometheus_image_prometheus="openshift/prometheus:v2.0.0-dev"
+openshift_prometheus_image_alertmanager="openshift/prometheus-alertmanager:dev"
+openshift_prometheus_image_alertbuffer="ilackarms/message-buffer"
+----
+
+Run the playbook:
+----
+$ ansible-playbook -vvv -i ${INVENTORY_FILE} playbooks/byo/openshift-cluster/openshift-prometheus.yml
+----
+
+[openshift-prometheus--additional-deploy]
+==== Additional Methods for Deploying Prometheus
+
+*Deploy Using Node-Selector*
+
+Label the node on which you want to deploy Prometheus:
+----
+# oadm label node/$NODE ${KEY}=${VALUE}
+----
+
+Deploy Prometheus with Ansible and container resources:
+----
+# Inventory file
+openshift_prometheus_state=present
+openshift_prometheus_namespace=prometheus
+openshift_prometheus_replicas=1
+
+# Set node selector for prometheus
+openshift_prometheus_node_selector={"${KEY}":"${VALUE}"}
```

Comment:
Do we need to show all the inventory options or should we just show this one?

Reply:
@jcantrill What do you think?

Reply:
In the course of the text, I dont think we need to show all the options but be able to refer them to a table of what is there.

Reply:
`openshift_prometheus_replicas=1` should be removed. This is no longer configurable since Prometheus moved to stateful set